modules/notification/NotificationStack.tsx

```diff
@@ -31,38 +39,108 @@ const styleSheet: ThemeStyleSheetFactory = theme => ({
   }
 });
 
-const NotifcationPresenter: React.FC<IProps & WithStylesProps> = ({
+const NotifcationStack: React.FC<IProps & WithStylesProps> = ({
   className,
   cx,
-  notifications = [],
+  displayDuration = 4000,
   styles,
-  timeout = 500,
-  handleMouseOver,
-  handleMouseOut,
+  transitionTimeout = 500,
   wrappedRef
 }) => {
+  const BaseNotification = useRegistry<INotificationProps>("Notification");
+  const [notifications, setNotifications] = useState<INotification[]>([]);
+  const [timeouts, setTimeouts] = useState<{ [notificationId: string]: NodeJS.Timeout }>({});
+  const notificationsRef = useRef(notifications);
+  notificationsRef.current = notifications;
```

Comment:
Is this needed? I've never seen this pattern

Reply:
Yes, in order to have the current vars from react state inside the timeouts.
I found this solution [here](https://upmostly.com/tutorials/settimeout-in-react-components-using-hooks) (that link also includes a link to a github issue)